cmd/juju/application/deploy.go

```diff
@@ -1363,6 +1375,8 @@ func (c *DeployCommand) maybeReadLocalBundle(ctx *cmd.Context) (deployFn, error)
 			bundleMachines:      c.BundleMachines,
 			bundleStorage:       c.BundleStorage,
 			bundleDevices:       c.BundleDevices,
+			controllerName:      controllerName,
```

Comment:
There are other places that _partial_ bundleDeploySpecs are constructed. I would recommend populating the missing bits in `deployBundle` instead as I did for `TargetModelUUID` to be sure.

Reply:
Fair!